data/src/file_upload_session.rs

```diff
@@ -301,15 +301,21 @@ impl FileUploadSession {
         debug_assert_le!(xorb.num_bytes(), *MAX_XORB_BYTES);
         debug_assert_le!(xorb.data.len(), *MAX_XORB_CHUNKS);
 
+        if xorb.num_bytes() > 0 {
+            // Add the xorb info to the current shard.
+            self.shard_interface.add_cas_block(xorb.cas_info.clone()).await?;
+
+            self.register_new_xorb_for_upload(xorb).await?;
```

Comment:
Let's drop the change below and move this call out of the if statement -- inside this function we already do:

```
         if xorb.num_bytes() == 0 {
            self.completion_tracker.register_xorb_upload_completion(xorb_hash).await;
            return Ok(true);
        }
```

And I'd like to keep the completion_tracker logic minimal